README.md

```diff
@@ -0,0 +1,76 @@
+# pthread lockdep
+
+`lockdep` is a deadlock detection tool designed for use with pthread_mutex. It is based on the original code from [http://ceph.com/dev-notes/lockdep-for-pthreads](http://ceph.com/dev-notes/lockdep-for-pthreads) and has been modified for enhanced functionality. This tool is distributed under the GNU Lesser General Public License version 2.1.
+
+## What is lockdep?
+
+`lockdep` is a software utility that helps developers detect potential deadlocks in their multithreaded applications. It works by tracking the locking dependencies between different mutexes in your application to identify circular dependencies that could lead to a deadlock.
+
+## Integrating lockdep into Your Project
+
+To integrate `lockdep` into your project, include the `lockdep.h` header file in your source code and link against the `lockdep` library when compiling your application. Ensure that you replace calls to `pthread_mutex_lock` and `pthread_mutex_unlock` with `mutex_lock` and `mutex_unlock` provided by `lockdep`.
+
+## Usage Instructions
+
+To use `lockdep`, initialize it in your application by including the `lockdep.h` header and replacing mutex lock and unlock calls:
+
+```c
+#include "lockdep.h"
+
+mutex_lock(&my_mutex);
+// Your critical section code here
+mutex_unlock(&my_mutex);
+```
+
+`lockdep` will automatically track all mutex locking operations and report potential deadlocks.
+
+## Examples
+
+Below is an example demonstrating how to use `lockdep` in a simple application:
+
+```c
+#include <pthread.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include "lockdep.h"
+
```

Comment:
```suggestion
#include <unistd.h>

```
fix build warning, refer to 
https://stackoverflow.com/questions/39156743/sleep-warning-implicit-declaration-of-function-sleep